emeis/core/models.py

```diff
@@ -1,7 +1,127 @@
-from django.contrib.auth.models import AbstractUser
+import unicodedata
+import uuid
 
-# from django.db import models
+from django.contrib.auth.validators import UnicodeUsernameValidator
+from django.contrib.postgres.fields import JSONField
+from django.db import models
+from django.utils import timezone
+from django.utils.translation import gettext_lazy as _
 
 
-class User(AbstractUser):
-    pass
+class BaseModel(models.Model):
+    created_at = models.DateTimeField(auto_now_add=True, db_index=True)
+    modified_at = models.DateTimeField(auto_now=True, db_index=True)
+    created_by_user = models.CharField(
+        max_length=150, blank=True, null=True, db_index=True
+    )
+    created_by_group = models.CharField(
+        max_length=150, blank=True, null=True, db_index=True
+    )
+
+    class Meta:
+        abstract = True
+
+
+class UUIDModel(BaseModel):
+    """
+    Models which use uuid as primary key.
+
+    Defined as Caluma default
+    """
+
+    id = models.UUIDField(primary_key=True, default=uuid.uuid4, editable=False)
+
+    class Meta:
+        abstract = True
+
+
+class User(UUIDModel):
```

Comment:
Will the API allow creating users with an already-known PK? For example if our IDP is Keycloak, will we allow the UUID to keep in sync or will we always use a separate UUID? Might be worth considering: In another project, we didn't do this and always have to think about if the "user id" is the remote or the local one". 

Doesn't help of course if the remote IDP uses anyhting other than UUID for identification. Still worth to consider though :)

Reply:
I don't see a reason why we should not allow setting the UUID.

Reply:
It's more about me being unsure if we can do this by default via DRF. Might be worth having a test for :)